taichi/ir/frontend_ir.h

```diff
@@ -170,17 +170,33 @@ class FrontendForStmt : public Stmt {
     }
   }
 
-  FrontendForStmt(const ExprGroup &loop_var, const Expr &global_var, Arch arch);
+  FrontendForStmt(const ExprGroup &loop_var,
+                  const Expr &global_var,
+                  Arch arch,
+                  int bit_vectorize,
```

Comment:
Could you actually create a struct to hold these params? That way `FrontendForStmt` and `ForLoopDecoratorRecorder` can share the struct, instead of repeating yourself. Thx!

Reply:
You mean that let FrontendForStmt holds ptr to ForLoopDecoratorRecorder?

Reply:
> You mean that let FrontendForStmt holds ptr to ForLoopDecoratorRecorder?

Oh, sorry, my bad. Holding ptr to ForLoopDecoratorRecorder is not feasible.

Reply:
Nah, I meant something like this:

```cpp

struct ForLoopConfig {
  int bit_vectorize;
  int num_cpu_threads;
  ...
};

class FrontendForStmt {
 public:
  FrontendForStmt(..., const ForLoopConfig &config);
};

class ForLoopDecoratorRecorder {
  ForLoopConfig config;
};
```

What do you think :-)

Reply:
> What do you think

Got it! But why not use `ForLoopConfig` directly if `ForLoopDecoratorRecorder` only wraps `ForLoopConfig` simply? The 
existence of the wrapper is convenient for us to construct/reset `ForLoopConfig`?

Reply:
Recorder sounds like something with behavior, whereas Config is just a struct. That said, sounds good to me if we just use recorder for simplicity. I'm not sure why "Holding ptr to ForLoopDecoratorRecorder is not feasible.", though.

Reply:
> Recorder sounds like something with behavior, whereas Config is just a struct. That said, sounds good to me if we just use recorder for simplicity. I'm not sure why "Holding ptr to ForLoopDecoratorRecorder is not feasible.", though.

"Holding ptr to ForLoopDecoratorRecorder" : My **old incorrect** thought is let `FrontendForStmt` and `ASTBuilder` share a `ForLoopDecoratorRecorder` instead of bit_vectorize, num_cpu_threads which are copy-saved by `FrontendForStmt`, which is not feasible because the `ForLoopDecoratorRecorder` which is held by ASTBuilder will be reset after constructing `FrontendForStmt`.

Reply:
@k-ye FINISHED. Please review #4309